### ISSUE TITLE
How about the ball ?

### ISSUE THREAD

**Issue by dovahkrid** [rjtavares/BirdsPyView#1]
I know this project is only for one-frame situation. But if I want to do a video frame-by-frame (low tech since I know nothing about OpenCV or something like that), then how about the ball position ?

And do you have any intention to do something about OpenCV which detect the position of the Penalty Box lines ?

**Comment by rjtavares** [comment]
Yes, I am considering adding an automatic detection component to the project (once I get the basics right).

Regarding the ball, it is traditionally a harder problem than detecting the players (one of the reasons being that the ball is frequently in the air, making it impossible to detect or infer its location based on only one perspective ).

How to deal with the ball is something I haven't decided yet, so I'll keep this issue open. Suggestions are welcomed.

**Comment by haz** [comment]
How accurate do you think the size of the ball could be measured? Given the resolution of the camera, this may be enough to determine the ball's distance along the ray from the camera to the field that the ball is occluding (and thus a prediction of the ball's location, air or otherwise).

**Comment by rjtavares** [comment]
With higher resolutions, that may be a cool solution. My pragmatic solution is to log touches/passes manually and extrapolate from there. With broadcast footage I think a mix of automatic and manual input is always necessary, because from a single POV there are always occlusions and objects out of view...

**Comment by haz** [comment]
Ah, fair point on the event-driven approach. I think any curve or non-linear path of the ball would be rare enough for this to be more reliable than ball distance based on pixel-counting the ball size.
